### ISSUE TITLE
RDF Export: Not a valid URI issue

### ISSUE THREAD

**Issue by tomas-knap** [odalic/sti#154]
Running task using the file and latest develop (7. 9. morning): 
[Books-v2.csv.zip](https://github.com/odalic/odalic-ui/files/458639/Books-v2.csv.zip)

When I see the feedback, I click RDF export:

![screen shot 2016-09-07 at 10 39 31](https://cloud.githubusercontent.com/assets/3014917/18305417/be048fe6-74e7-11e6-88d1-e8f820911b14.png)

As a result, I got: 

![screen shot 2016-09-07 at 10 39 20](https://cloud.githubusercontent.com/assets/3014917/18305403/b5b02382-74e7-11e6-9061-13f1f27cf7ea.png)

Solution: It should provide some human readable message, why the error failed. Connected with improved REST API, with better error messages.

**Comment by tomas-knap** [comment]
CSV/JSON export works

**Comment by tomas-knap** [comment]
@brodecva If Josef can fix that (test it on that Books example) that would be great

**Comment by tomas-knap** [comment]
Could be connecting with encoding issues, because CSV export has strange chars: 

```
Book;Author;Series;Book_url;Book_alternative_urls;Author_url;Author_alternative_urls;Series_url;Series_alternative_urls
Gardens of the Moon;Steven Erikson;Malazan Book of the Fallen;http://dbpedia.org/resource/Gardens_of_the_Moon;;http://dbpedia.org/resource/Steven_Erikson;http://de.dbpedia.org/resource/Steven_Erikson;http://dbpedia.org/resource/Category:Malazan_Book_of_the_Fallen;
Deadhouse Gates;Steven Erikson;Malazan Book of the Fallen;http://dbpedia.org/resource/Deadhouse_Gates;;http://dbpedia.org/resource/Steven_Erikson;http://de.dbpedia.org/resource/Steven_Erikson;http://dbpedia.org/resource/Category:Malazan_Book_of_the_Fallen;
Jost Friedrich SchÃ¤ffer;Franz BÃ¶sken;Acta Organologica;http://www.wikidata.org/entity/Q1708791;http://de.dbpedia.org/resource/Jost_Friedrich_SchÃ¤ffer;http://www.wikidata.org/entity/Q15437644;http://de.dbpedia.org/resource/Franz_BÃ¶sken;http://www.wikidata.org/entity/Q343172;http://de.dbpedia.org/resource/Acta_Organologica
```

Or it could be connected with export of relations (which are not exported in CSV)

**Comment by brodecva** [comment]
Both I and @josef-janousek cannot reproduce the error. Not even in Chrome. Yes, the encoding of the export might be scrambled as the browser or your editor cannot detect it is in Unicode, but that should not affect the server, which should use Unicode internally.
**What would help us to debug it, is the following: Could you please make the server break at the point of the thrown exception and send us the string it fails to make the IRI from? Otherwise we have nothing to work with. Thanks!**

**Comment by josef-janousek** [comment]
Seems that these strange chars are also in the original input file Books-v2.csv ...

**Comment by brodecva** [comment]
>Seems that these strange chars are also in the original input file Books-v2.csv ...

Still convinced it is false flag, as you mentioned on Skype, only some programs show it scrambled.

**Comment by tomas-knap** [comment]
So I was debugging that,  the problem is the empty value "" when new triple is created:

<img width="762" alt="screen shot 2016-09-20 at 19 57 49" src="https://cloud.githubusercontent.com/assets/3014917/18682688/804ffa10-7f6d-11e6-87c1-b03eec5934ff.png">

For sure in case of this triple, so in case there is no alternative url for disambiguation:

<img width="552" alt="screen shot 2016-09-20 at 19 56 40" src="https://cloud.githubusercontent.com/assets/3014917/18682767/cbddb800-7f6d-11e6-92fa-6b29b4701e18.png">

**Comment by brodecva** [comment]
@tomas-knap Thank you!

@josef-janousek Do you have some explanation for the fact that the "item" is empty?

**Comment by josef-janousek** [comment]
@tomas-knap Also thank you for debugging.

I did not reproduce the situation when the "item" is empty (with the file Books-v2.csv). As depicted on the screenshot, all items added by this part of code were not empty.
I found that the "validation" condition in the setIRIString method of the SimpleIRI class in the RDF4J library, which throws this exception, is: 
`if (iriString.indexOf(':') < 0) {
	throw new IllegalArgumentException("Not a valid (absolute) IRI: " + iriString);
}`
So of course I could add this validation also to our DefaultAnnotatedTableToRDFExportAdapter class (probably to all parts of code where the IRI is created using the createIRI method from RDF4J). Is it OK?
![rdf-export](https://cloud.githubusercontent.com/assets/17772934/18731517/b7d183ec-805c-11e6-9a24-9989ab845de3.png)
